src/essentialsTP/essentialsTP.php

```diff
@@ -5,1373 +5,1191 @@
+					$this->prepare->bindValue(":time", time(), SQLITE3_INTEGER);
+					$this->prepare->bindValue(":name", $name, SQLITE3_TEXT);
+					$this->result = $this->prepare->execute();
+					return true;
+				default:
+					return false;
+			}
+		}
+	}
+
+	public function onPlayerDeath(PlayerDeathEvent $event)
+	{
+		$player = $event->getEntity();
+		$this->death_loc[$player->getName()] = new Position(
+			round($player->getPosition()->getX()),
+			round($player->getPosition()->getY()),
+			round($player->getPosition()->getZ()),
+			$player->getWorld()
+		);
+	}
+
+	public function onPlayerSleep(PlayerBedEnterEvent $event)
+	{
+		if ($this->config->get("bed-sets-home") == true) {
+			$player = $event->getPlayer();
+			if ($player->hasPermission("essentialstp.command.bedsethome")) {
+				$this->player_cords = array('x' => (int) $player->getPosition()->getX(), 'y' => (int) $player->getPosition()->getY(), 'z' => (int) $player->getPosition()->getZ());
+				$this->username = $player->getName();
+				$this->world = $player->getWorld()->getFolderName();
+				$this->home_loc = "bed";
+				$this->prepare = $this->db2->prepare("SELECT player,title,x,y,z,world FROM homes WHERE player = :name AND title = :title");
+				$this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
+				$this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
+				$this->result = $this->prepare->execute();
+				$sql = $this->fetchall();
+				if (count($sql) > 0) {
+					$this->prepare = $this->db2->prepare("UPDATE homes SET world = :world, title = :title, x = :x, y = :y, z = :z WHERE player = :name AND title = :title");
+					$this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
+					$this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
+					$this->prepare->bindValue(":world", $this->world, SQLITE3_TEXT);
+					$this->prepare->bindValue(":x", $this->player_cords['x'], SQLITE3_TEXT);
+					$this->prepare->bindValue(":y", $this->player_cords['y'], SQLITE3_TEXT);
+					$this->prepare->bindValue(":z", $this->player_cords['z'], SQLITE3_TEXT);
+					$this->result = $this->prepare->execute();
+				} else {
+					$this->prepare = $this->db2->prepare("INSERT INTO homes (player, title, world, x, y, z) VALUES (:name, :title, :world, :x, :y, :z)");
+					$this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
+					$this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
+					$this->prepare->bindValue(":world", $this->world, SQLITE3_TEXT);
+					$this->prepare->bindValue(":x", $this->player_cords['x'], SQLITE3_TEXT);
+					$this->prepare->bindValue(":y", $this->player_cords['y'], SQLITE3_TEXT);
+					$this->prepare->bindValue(":z", $this->player_cords['z'], SQLITE3_TEXT);
+					$this->result = $this->prepare->execute();
+				}
+			}
+		}
+	}
+
+	public function onBlockTap(PlayerInteractEvent $event)
+	{
+		if ($event->isCancelled())
```

Comment:
This is no longer required since API 4. @handleCancelled is already the default.